packages/ui/src/pages/Enable.ts

```diff
@@ -0,0 +1,216 @@
+  let sessionLedgers;
+  let ddClass: string = 'dropdown';
+
+  store.getAvailableLedgers((availableLedgers) => {
+    if (!availableLedgers.error) {
+      sessionLedgers = availableLedgers;
+    }
+  });
+  
+  if (active) ddClass += ' is-active';
+  const flip = () => {
+    setActive(!active);
+  };
+
+  const setLedger = (ledger) => {
+    store.setLedger(ledger);
+    flip();
+  };
+
+  useEffect(() => {
+    // eslint-disable-next-line no-unused-vars
+    chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
+      if (request.body.method == JsonRpcMethod.EnableAuthorization) {
+        // Set the request in the store with origin so we can respond later
+        setRequest(request);
+        store.saveRequest(request);
+        responseOriginTabID = request.originTabID;
+
+        // Check for existence of the params and set page values
+        if (request.body.params.promptedAccounts && request.body.params.promptedAccounts.length > 0) {
+          setPromptedAccounts(request.body.params.promptedAccounts);
+        }
+        if (request.body.params.genesisId) {
+          setGenesisId(request.body.params.genesisId);
+        }
+        if (request.body.params.genesisHash) {
+          setGenesisHash(request.body.params.genesisHash);
+        }
+        if (request.body.params.ledger) {
+          // Ledger is added during EnableAuthorization to match with legacy ledger name
+          store.setLedger(request.body.params.ledger);
+        }
+      }
+    })
+  }, []);
+
+  const grant = () => {
+    window.removeEventListener('beforeunload', deny);
+    chrome.runtime.sendMessage({
+      source: 'extension',
+      body: {
+        jsonrpc: '2.0',
+        method: JsonRpcMethod.AuthorizationAllow,
+        params: {
+          responseOriginTabID: responseOriginTabID,
+          isEnable: true,
+          genesisId: genesisId,
+          genesisHash: genesisHash, 
+          accounts: accounts,
+          ledger: store.ledger
+        },
+      },
+    });
+  };
+
+  return useObserver(
+    () => html`
+    <div class="main-view" style="flex-direction: column; justify-content: space-between;">
+      <div class="px-4 mt-2" style="flex: 0; border-bottom: 1px solid #EFF4F7">
+        <img src=${logotype} width="130" />
+      </div>
+      <div style="flex: 1">
+        <section class="hero">
+          <div class="hero-body py-5">
+            ${request.favIconUrl &&
+            html` <img src=${request.favIconUrl} width="48" style="float:left" /> `}
+            <h1 class="title is-size-4" style="margin-left: 58px;">
+              Access requested to your wallet${request.originTitle && html` from ${request.originTitle}`}
+            </h1>
+          </div>
+        </section>
+        <section class="px-2 py-0">
+          <h3> Select ledger and accounts to share${request.originTitle && html` with ${request.originTitle}`}. <b>Bolded</b> accounts are required by the dApp. </h3>
```

Comment:
Using is required because it will result in a failure if not included. Requested would indicate that a successful approval could occur without all of those accounts being shared which is not the case.